org.jcryptool.analysis.friedman/src/org/jcryptool/analysis/friedman/ui/FriedmanGraphUI.java

```diff
@@ -94,7 +101,23 @@ public void widgetSelected(SelectionEvent e) {
 				public void update(Observable o, Object arg) {
 					if (textSelector.getText() != null) {
 						if (textSelector.getText().getText() != null) {
+							if(textSelector.getText().getText().length() < friedman_max_text_length){
 							message = textSelector.getText().getText();
+							System.out.println(message.length());
+							lastSuccessfullLoadedText = textSelector.getText();
+							}
+							else{
+								Shell shell2 = new Shell();

```

Comment:
Do not use this; use instead of `shell2`: `FriedmanGraphUI.this.getDisplay().getActiveShell()`
see comment **I2)**